c7n/resources/dms.py

```diff
@@ -224,3 +238,199 @@ def process_resource_set(self, resources, tags):
+            'certificatearn': {'type': 'string'},
+            'databasename': {'type': 'string'},
+            'endpointidentifier': {'type': 'string'},
+            'extraconnectionattributes': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'dynamodbsettings': {
+                'type': 'object',
+                'additionalProperties': False,
+                'properties': {'serviceaccessrolearn': {'type': 'string'}}
+            },
+            's3settings': {
+                'type': 'object',
+                'additionalProperties': False,
+                'properties': {
+                    'bucketfolder': {'type': 'string'},
+                    'bucketname': {'type': 'string'},
+                    'compressiontype': {
+                        'type': 'string', 'enum': ['none', 'gzip']
+                    },
+                    'csvdelimiter': {'type': 'string'},
+                    'csvrowdelimiter': {'type': 'string'},
+                    'externaltabledefinition': {'type': 'string'},
+                    'serviceaccessrolearn': {'type': 'string'}
+                }
+            },
+            'mongodbsettings': {
+                'type': 'object',
+                'additionalProperties': False,
+                'properties': {
+                    'authmechanism': {
+                        'type': 'string', 'enum': [
+                            'default', 'mongodb_cr', 'scram_sha_1']
+                    },
+                    'username': {'type': 'string'},
+                    'password': {'type': 'string'},
+                    'databasename': {'type': 'string'},
+                    'docstoinvestigate': {'type': 'integer', 'minimum': 1},
+                    'extractdocid': {'type': 'boolean'},
+                    'nestinglevel': {
+                        'type': 'string', 'enum': ['none', 'one']},
+                    'port': {
+                        'type': 'integer', 'minimum': 1, 'maximum': 65535},
+                    'servername': {'type': 'string'}
+                }
+            }
+        }
+    }
+    permissions = ('dms:ModifyEndpoint',)
+
+    def configure_s3_params(self, e, params):
+        settings = self.data.get('s3settings')
+        if not settings:
+            raise KeyError('s3settings not provided')
+
+        params['S3Settings'] = {
+            'BucketName': settings.get(
+                'bucketname', e['S3Settings']['BucketName']),
+            'CsvDelimiter': settings.get(
+                'csvdelimiter', e.get('S3Settings', {}).get(
+                    'CsvDelimiter')) or ',',
+            'CsvRowDelimiter': settings.get(
+                'csvrowdelimiter', e.get('S3Settings', {}).get(
+                    'CsvRowDelimiter')) or '\n',
+            'CompressionType': settings.get(
+                'compressiontype', e.get('S3Settings', {}).get(
+                    'CompressionType')) or 'none'}
```

Comment:
botodocs say default is NONE, is none (lowercase) the same?

Reply:
yes. NONE/ONE or none/one... case does not appear to be relevant but having said that, i might need to add a `lower()` to ensure it's uniform